src/librustc/ty/maps.rs

```diff
@@ -96,9 +96,9 @@ impl<'tcx> Value<'tcx> for Ty<'tcx> {
     }
 }
 
-pub struct CycleError<'a, 'tcx: 'a> {
+pub struct CycleError<'tcx> {
     span: Span,
-    cycle: RefMut<'a, [(Span, Query<'tcx>)]>,
+    cycle: Vec<(Span, Query<'tcx>)>,
```

Comment:
Why? This is going to clone all over the place now in early coherence and whatnot.